From 8abee208e8aef9c2f230de2247a455f4b2a2896b Mon Sep 17 00:00:00 2001
Message-Id: <8abee208e8aef9c2f230de2247a455f4b2a2896b.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vijay Purushothaman <vijay.a.purushothaman@intel.com>
Date: Tue, 9 Jul 2013 20:20:52 +0530
Subject: [PATCH 034/312] REVERTME [VPG]: drm/i915: Don't switch off the eDP
 panel power for init sequence

This is a temporary change for BYT.
For DP/eDP aux transcations to succeed we need only the VDD.
Some of the FFRD10 PR1.1B boards (most of the board works fine without this
hack) need both the panel power & vdd to be ON for aux transactions to
succeed. This could be a board problem. Till we figure out what's going on,
use this temporary hack.

(cherry picked from commit adb4ca559f649722e0cc5b95a817e9b2a72a92a9)

Issue:APDEV-1165
Change-Id: I5bb88f88d1f929c903185739ec3a04d50c0b6c2d
Signed-off-by: Vijay Purushothaman <vijay.a.purushothaman@intel.com>
Reviewed-on: http://android.intel.com:8080/118673
Reviewed-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Tested-by: Sun, Hang L <hang.l.sun@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c   |    5 ++++-
 drivers/gpu/drm/i915/intel_hdmi.c |    1 -
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index ce7097f..3d9f8b0 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1877,7 +1877,10 @@ static void intel_disable_dp(struct intel_encoder *encoder)
 	intel_edp_panel_vdd_on(intel_dp);
 	intel_edp_backlight_off(intel_dp);
 	intel_dp_sink_dpms(intel_dp, DRM_MODE_DPMS_OFF);
-	intel_edp_panel_off(intel_dp);
+
+	/* Some of the FFRD10 PR1.1B boards doesnt like when edp panel power
+	 * is off */
+	/* intel_edp_panel_off(intel_dp); */
 
 	/* cpu edp my only be disable _after_ the cpu pipe/plane is disabled. */
 	if (!(port == PORT_A || IS_VALLEYVIEW(dev)))
diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 891b621..ff03a8d 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -784,7 +784,6 @@ static void intel_enable_hdmi(struct intel_encoder *encoder)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct intel_crtc *intel_crtc = to_intel_crtc(encoder->base.crtc);
 	struct intel_hdmi *intel_hdmi = enc_to_intel_hdmi(&encoder->base);
-	struct intel_crtc *crtc = to_intel_crtc(encoder->base.crtc);
 	u32 temp;
 	u32 enable_bits = SDVO_ENABLE;
 
-- 
1.7.9.5

